datafusion/proto/proto/datafusion.proto

```diff
@@ -825,17 +828,6 @@ enum PrimitiveScalarType{
     TIME64 = 27;
 }
 
-message ScalarType{
```

Comment:
I am not sure why but the previous code was encoding `ScalarValue::List` using a different pattern than the struct. This PR changes the serialization code to use the same structure as `ScalarValue::List` -- the Field and the a list of scalar values